spec/ruby/core/range/bsearch_spec.rb

```diff
@@ -126,8 +126,8 @@
         inf = Float::INFINITY
         (0..inf).bsearch { |x| x == inf }.should == inf
         (0...inf).bsearch { |x| x == inf }.should == nil
-        (-inf..0).bsearch { |x| x == -inf }.should == nil
-        (-inf...0).bsearch { |x| x == -inf }.should == nil
+        (-inf..0).bsearch { |x| x != -inf }.should == -Float::MAX
+        (-inf...0).bsearch { |x| x != -inf }.should == -Float::MAX
         (inf..inf).bsearch { |x| true }.should == inf
```

Comment:
I think those are actually correct, and should be kept (in addition to the new ones). Ditto for those below in this file.

Reply:
For bsearch the block must be either monotonically false ... true or monotonically 1 ... 0 ... -1 . (https://ruby-doc.org/core-2.7.2/Range.html#method-i-bsearch explains it better than me). The blocks I replaced were monotonically the other way around, like in this case -inf is true and everything larger is false; so I just flipped the values.

Reply:
Forgot to answer this. Okay, makes sense. I was coming from the perspective that these tests work on MRI, but I think it's fair that nobody is going to rely on this.